sys-cluster/keepalived/keepalived-2.2.1.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools systemd
+
+DESCRIPTION="A strong & robust keepalive facility to the Linux Virtual Server project"
+HOMEPAGE="https://www.keepalived.org/"
+SRC_URI="https://www.keepalived.org/software/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sparc ~x86"
+IUSE="+bfd dbus -json regex snmp systemd"
+
+RDEPEND="dev-libs/libnl:=
+	dev-libs/openssl:=
+	dev-libs/popt
+	net-libs/libnfnetlink
+	sys-apps/iproute2
+	regex? ( >=dev-libs/libpcre2-8 )
+	dbus? ( sys-apps/dbus dev-libs/glib:2 )
+	json? ( dev-libs/json-c:= )
+	snmp? ( net-analyzer/net-snmp )"
+DEPEND="${RDEPEND}
+	>=sys-kernel/linux-headers-4.4"
+
+DOCS=(
+	README CONTRIBUTORS INSTALL ChangeLog AUTHOR TODO
+	doc/keepalived.conf.SYNOPSIS doc/NOTE_vrrp_vmac.txt
+)
+
+src_prepare() {
+	default
+
+	eautoreconf
+}
+
+src_configure() {
+	local init
+	if use systemd; then
+		init=systemd
+	else
+		init=custom
+	fi
+
+	# keepalived has support to dynamically use some libraries instead of
+	# linking them:
+	#--enable-dynamic-linking \
+	#--enable-libiptc-dynamic \
+	#--enable-libnl-dynamic \
+	#--enable-libxtables-dynamic \
+	STRIP=/bin/true \
+	econf \
+		--with-init=${init} \
+		--with-kernel-dir=/usr \
+		--enable-sha1 \
+		--enable-vrrp \
+		$(use_enable bfd) \
+		$(use_enable dbus) \
+		$(use_enable json) \
+		$(use_enable regex) \
+		$(use_enable snmp) \
+		$(use_enable snmp snmp-rfc) \
+		$(use_enable systemd)
+}
+
+src_install() {
+	default
+
+	newinitd "${FILESDIR}"/keepalived.init-r1 keepalived
+	newconfd "${FILESDIR}"/keepalived.confd-r1 keepalived
+
+	systemd_newunit "${FILESDIR}"/${PN}.service ${PN}.service
```

Comment:
It looks like enabling systemd will install upstream service files: https://github.com/acassen/keepalived/blob/v2.2.1/keepalived/Makefile.am#L68-L70

I am also not a systemd user but setting correct `Type` is important now that keepalived-2.2.x supports notify, https://github.com/acassen/keepalived/commit/027157a0617bea626723af5e32d8fe6ca53855b5

@gentoo/systemd Can you help? What do you suggest? Upstream would install https://github.com/acassen/keepalived/blob/v2.2.1/keepalived/keepalived.service.in and we are currently using https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-cluster/keepalived/files/keepalived.service?id=2c77e25783da444692e66820b72472c21bbeab32

Reply:
I've updated ours, that way ``/etc/systemd/system/keepalived.service.d/00gentoo.conf`` is present for customization and this is omitted: ``EnvironmentFile=-/etc/sysconfig/keepalived``